homeassistant/components/camera/services.yaml

```diff
@@ -51,6 +51,22 @@ play_stream:
       description: (Optional) Stream format supported by media player.
       example: 'hls'
 
+record:
+  description: Record live camera feed.
+  fields:
+    entity_id:
+      description: Name of entities to record.
+      example: 'camera.living_room_camera'
+    filename:
+      description: Template of a Filename. Variable is entity_id.  Must be mp4.
+      example: '/tmp/snapshot_{{ entity_id }}.mp4'
+    duration:
+      description: (Optional) Target recording length (in seconds).
+      example: 30
+    lookback:
+      description: (Optional) Target lookback period (in seconds) to include in addition to duration.
```

Comment:
Does this always work ? Or does it require something.

Reply:
It will not always work, but is handled properly.  If there is no active HLS stream, this value will be ignored.  If the value is set to a crazy number like 10000, it will include all of the HLS segments we have currently.  This is the formula: `min(int(lookback // hls.target_duration), hls.num_segments)`

Reply:
Let's add it to the description.